Fix logic to determine master vs PR

driazati opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65157
* **#65155**

This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

Test plan: see `[scribe] Skipping RDS write on PR` in the logs for `linux-xenial-cuda11.3-py3.6-gcc7`

Differential Revision: [D31029048](https://our.internmc.facebook.com/intern/diff/D31029048)

driazati committed:
Fix logic to determine master vs PR

This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

[ghstack-poisoned]

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

[ghstack-poisoned]

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

[ghstack-poisoned]

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

[ghstack-poisoned]

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

TODO: Look through the logs and ensure `binary_size` is not being written to

[ghstack-poisoned]

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

TODO: Look through the logs and ensure `binary_size` is not being written to

[ghstack-poisoned]

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

Test plan: see `[scribe] Skipping RDS write on PR` in the logs for `linux-xenial-cuda11.3-py3.6-gcc7`

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029048).

malfet commented on tools/stats/scribe.py:
```diff
@@ -24,9 +27,7 @@ def aws_lambda() -> Any:
 
 
 def invoke_lambda(name: str, payload: Any) -> Any:
-    res = aws_lambda().invoke(
-        FunctionName=name, Payload=json.dumps(payload).encode()
-    )
+    res = aws_lambda().invoke(FunctionName=name, Payload=json.dumps(payload).encode())
```
Q: Can we somehow avoid change those back-n-forth?

driazati committed:
Update on "Fix logic to determine master vs PR"


This was bugged before on empty strings which caused the hook to write on any job, not just `master` regardless of the `only_on_master` flag.

Test plan: see `[scribe] Skipping RDS write on PR` in the logs for `linux-xenial-cuda11.3-py3.6-gcc7`

Differential Revision: [D31029048](https://our.internmc.facebook.com/intern/diff/D31029048)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029048).

seemethere commented on .github/templates/linux_ci_workflow.yml.j2:
```diff
@@ -171,6 +171,7 @@ jobs:
           CIRCLE_SHA1: ${{ github.event.pull_request.head.sha || github.sha }}
           CIRCLE_TAG: ${{ steps.parse-ref.outputs.tag }}
           CIRCLE_WORKFLOW_ID: '${{ github.run_id }}_${{ github.run_number }}'
+          GITHUB_WORKFLOW_RUN_ID: '${{ github.run_id }}'
```
Isn't this already a thing?

`GITHUB_RUN_ID` should give you the same info and it's already built in 

https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables
